react-app/src/store/reviews.js

```diff
@@ -0,0 +1,105 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const initialState = { entities: { reviews: {} } }
+
+export const createReview = createAsyncThunk(
+    'reviews/createReview',
+    async (itemId, formDetails, thunkAPI) => {
+        const response = await fetch(`/api/items/${itemId}/reviews`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(formDetails)
+        })
+        const data = await response.json();
+        if (response.ok && !data.errors) {
+            return data;
+        } else if (response.status < 500) {
+            throw thunkAPI.rejectWithValue(data.errors);
+        } else {
+            throw thunkAPI.rejectWithValue(['An error occured. Please try again.'])
+        }
+    }
+)
+
+export const getReviews = createAsyncThunk(
+    'reviews/getReviews',
+    async (itemId, thunkAPI) => {
+        const response = await fetch(`/api/items/${itemId}/reviews`, {
+            headers: {
+                "Content-Type": "application/json",
+            },
+        });
+        const data = await response.json();
+        if (response.ok) {
+            return data;
+        } else {
+            throw thunkAPI.rejectWithValue(["An error occurred. Please try again."]);
+        }
+    }
+)
+
+export const editReview = createAsyncThunk(
+    'reviews/editReview',
+    async (reviewId, formDetails, thunkAPI) => {
```

Comment:
The parameters `reviewId` and `formDetails` need to be in an object.
Change this to `({reviewId, formDetails}, thunkAPI)`